cmd/sdk-server/main.go

```diff
@@ -128,28 +108,101 @@ func main() {
 
 		var s *gameservers.SDKServer
 		s, err = gameservers.NewSDKServer(viper.GetString(gameServerNameEnv), viper.GetString(podNamespaceEnv),
-			healthDisabled, healthTimeout, healthFailureThreshold, healthInitialDelay, kubeClient, agonesClient)
+			ctlConf.HealthDisabled, ctlConf.HealthTimeout, ctlConf.HealthFailureThreshold,
+			ctlConf.HealthInitialDelay, kubeClient, agonesClient)
 		if err != nil {
 			logger.WithError(err).Fatalf("Could not start sidecar")
 		}
-		ctx, cancel := context.WithCancel(context.Background())
-		defer cancel()
 
 		go s.Run(ctx.Done())
 		sdk.RegisterSDKServer(grpcServer, s)
 	}
 
+	run(ctx, grpcServer, lis, grpcEndpoint, mux, httpServer)
+
+	<-stop
+	logger.Info("shutting down sdk server")
+}
+
+// run runs all the services
+func run(ctx context.Context, grpcServer *grpc.Server, lis net.Listener, grpcEndpoint string,
```

Comment:
One way to get around all the arguments to `run` would be to introduce a type, maybe `sdkServer` and attach `run` to it. But again, style. Not a big deal.